src/extension.ts

```diff
@@ -81,6 +81,9 @@ export function activate(context: vscode.ExtensionContext) {
     context.subscriptions.push(vscode.commands.registerCommand(
         `${SETTINGS_VIEW_NAME}.close-instance`,
         async (target: CDPTarget) => {

```

Comment:
Setting `target` as an optional parameter would catch this bug at transpile time (assuming the flakiness was caused by an NFE on line 73).  

Were you able to root cause the scenario where `target` is null or undefined? We may need to make other functions that rely on targets to assume they're optional as well (i.e line 70).

Reply:
Ok, I'll make it an optional parameter (the flakiness is from NPE on line 90 though). I'm not exactly sure what the root cause of the undefined argument is yet, but the scenario is:
1. click the close button
2. the extension takes some time to process this, it shows the blue clock icon and animation at the top of the targets sidepane
3. click the close button while the extension is still processing

Reply:
My bad, I typoed the error line. I meant line 90, where we access a property on the potentially null target.

Reply:
Line 70 seems to be the only other part similar to the one causing this issue. Do you think I should change that function to have an optional param too? I think that one's a lot less likely to cause an issue, but maybe it's an edge case

Reply:
In regards to line 70, the scenario is the extension attaching to the target. If there's a case where an attach fails (I'm not sure what would cause this), we could get an undefined target.

Double check with Michael and Vidal since they know the extension lifecycle better than I do.

Reply:
Got it, @mliao95 what do you think?

Reply:
Correct me if I'm wrong, but the issue here is that we get an undefined target when we spam the `close-instance` button because the first call to `close-instance` destroys the target and before the first call can update the UI, the second+ calls are attempting to delete the already deleted target.

If that is the case, disabling the `close-instance` and `attach` buttons should prevent either command to be called until the first call to `close-instance` is finished and we can assume that the CDPTarget will not be undefined.  

If we think that there is another issue causing the `undefined` target, I would go with making both `attach` and `close-instance` take optional parameters and continue investigating.